src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/controller/IdentifyingMarksController.kt

```
@@ -209,4 +210,51 @@ class IdentifyingMarksController(private val identifyingMarksService: Identifyin
     @Valid
     identifyingMarkUpdateRequest: IdentifyingMarkUpdateRequest,
   ): IdentifyingMarkDto = identifyingMarksService.update(uuid, identifyingMarkUpdateRequest)
+
+  @PutMapping("/mark/{uuid}/photo", produces = [MediaType.APPLICATION_JSON_VALUE])
+  @ResponseStatus(HttpStatus.OK)
+  @PreAuthorize("hasRole('ROLE_PRISON_PERSON_API__PRISON_PERSON_DATA__RW')")
+  @Operation(
+    summary = "Add a new photo for the identifying mark",
+    description = "Stores a new identifying mark photo and sets it to the current photo for the identifying mark." +
+      "The image file supplied on the file attribute of a multipart/form-date submission.  \n" +
+      "Returns the identifying mark detail\n" +
+      "Requires role `ROLE_PRISON_PERSON_API__PRISON_PERSON_DATA__RW`",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Returns uploaded photograph document data",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires ROLE_PRISON_PERSON_API__PRISON_PERSON_DATA__RW",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Data not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun updateIdentifyingMarkPhoto(
+    @Schema(description = "The UUID of the mark", example = "3946f7d9-25d0-449f-bf17-1ade41559391", required = true)
+    @PathVariable
+    uuid: String,
```

Worth making this a UUID type here so it fails early if it's not a valid UUID? I think Spring has a converter out of the box.  Or else some validation on the string at least so it's a bad request rather than a server error if it blows up later.

---

I didn't know this was a built in thing! I'll update the other instances as well